### PR TITLE
Feat/lesq 962/combined unit listing pages [LESQ-962]

### PR DESCRIPTION
**Pull request by njblore**
## Description

Music year: 1994

- remove custom logic for maths on subject listing and unit listing pages and promote for all subjects
- remove legacy flag from subject listing page query and simplify
- refactor subject card double to just a single
- refactor props for subject card
- refactor unit listing page and add tests for helper methods

## How to test

note: ks4 science subjects are a bit unique, we are going to remove the singular legacy units as they are replicated in the 'new' curriculum, this means there are technically programmes which are unreachable through this new journey but that is fine (also the counts for the unit and lessons will be incorrect on the combined listing until those units are taken down)

1. Go to subject listing pages eg https://deploy-preview-2647--oak-web-application.netlify.thenational.academy/teachers/key-stages/ks2/subjects
3. You should see all subjects are combined into a single card
4. EYFS should be all legacy
5. Specialist should be unaffected  https://deploy-preview-2647--oak-web-application.netlify.thenational.academy/teachers/specialist/subjects
6. Subjects with programmes should proceed to correct programme listing page
7. Unit listing page should correctly group legacy and new units

## Screenshots

How it used to look (delete if n/a):
Screenshot 2024-07-30 at 09 10 02
Screenshot 2024-07-30 at 09 10 49

How it should now look:

Screenshot 2024-07-30 at 09 10 30
Screenshot 2024-07-30 at 09 10 59

## Checklist

- [ ] Only display ‘new’ threads
- [ ] Add link for new/legacy curriculum pages
- [ ] Change subject card to be one button

### CODE REVIEW

**Review by RBrownsett — CHANGES_REQUESTED** [review]
@JBR90 couple of issues from initial testing:

1. Specialised subjects - the new unit card has bee applied but the formatting looks incorrect without padding:
image
2. The year identifier is missing from the top of the unit card that has optionality units (should be just to the right of the the number of unit options info) 
image
3. The disabled functionality for the unit cards seems not be working. Units that should be 'greyed out' and disabled can be clicked on and you are taken to the lesson listing page. Art KS3 example below - unit one should be greyed out: 
PR:
image

Live:
image

File: src/components/TeacherComponents/UnitList/helpers.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  legacy: boolean,
+) => {
+  return isCurrentPageItems(pageItems)
+    ? pageItems.filter((item) =>
+        legacy
+          ? isSlugLegacy(item[0]!.programmeSlug)
+          : !isSlugLegacy(item[0]!.programmeSlug),
+      )
+    : [];
+};
```
Comment (carlmw) [inline src/components/TeacherComponents/UnitList/helpers.tsx:36]: 🏄🏻 🤯 reading this broke me for a few reads. 

The argument name `legacy` obscures the intention (which I think is to let you filter out lessons with legacy slugs or vice versa).

The nested ternary hurt too. Could be refactored as an early return? 

```suggestion
export const getPageItems = (
  pageItems: CurrentPageItemsProps[] | SpecialistUnit[][],
  pickLegacyItems: boolean,
) => {
  if (!isCurrentPageItems(pageItems)) {
    return [];
  }

  return pageItems.filter((item) => {
    const isLegacy = isSlugLegacy(item[0]!.programmeSlug);
    return pickLegacyItems ? isLegacy : !isLegacy;
  });
};
```

**Review by carlmw — COMMENTED** [review]
Nothing blocking from a code perspective IMO. Happy to do another pass on any new commits once you've taken care of @RBrownsett's comments!

**Comment by RBrownsett** [comment]
@JBR90 another odd bug - navigating to an optionality unit then going from the lesson listing page back to the unit listing page the units are duplicated below the footer and we end up with two footers - see video below for behaviour (example KS4 English AQA)

https://github.com/user-attachments/assets/506bcff1-f721-4bcb-bdb7-45333e1be57a

File: src/components/TeacherComponents/UnitList/UnitList.tsx [hunk 2]
```diff
@@ -61,6 +62,37 @@ const isUnitListData = (
...
 };
 
+const getOptionalityUnits = (
+  unit: CurrentPageItemsProps | SpecialistUnit[],
+  onClick: (props: UnitListItemProps | SpecialistListItemProps) => void,
+  router: NextRouter,
+) => {
+  return unit.map((unitOption) => {
+    const handleClick = (e: MouseEvent<HTMLElement>) => {
+      // Tracking data was not being sent to avo before the page reload, so we prevent the default and use router to navigate to the page after the onClick
```
[the remark this answers is not in the document]
Comment (JBR90) [reply src/components/TeacherComponents/UnitList/UnitList.tsx:72]: Thanks, will give it a read for next time

**Comment by RBrownsett** [comment]
@JBR90 approved - changes have been made, great work! Testing notes detailed on the Notion ticket

**Pull request merged**